torch/optim/optimizer.py

```diff
@@ -179,17 +201,18 @@ def zero_grad(self, set_to_none: bool = False):
                 (in one case it does the step with a gradient of 0 and in the other it skips
                 the step altogether).
         """
-        for group in self.param_groups:
-            for p in group['params']:
-                if p.grad is not None:
-                    if set_to_none:
-                        p.grad = None
-                    else:
-                        if p.grad.grad_fn is not None:
-                            p.grad.detach_()
+        with torch.autograd.profiler.record_function(self._zero_grad_profile_name):
```

Comment:
one last thing - it turns that is some cases _zero_grad_profile_name doesn't exist here, could you add a check and initialize it here (potentially moving `self._zero_grad_profile_name = "Optimizer.zero_grad#{}.zero_grad".format(self.__class__.__name__)`) into a helper function?

Reply:
actually just calling `self._hook_for_profile()` if `self._zero_grad_profile_name` doesn't exist would suffice